beacon/blockchain/execution_engine.go

```diff
@@ -22,14 +22,15 @@ package blockchain
 
 import (
 	"context"
-	"time"
 
-	payloadtime "github.com/berachain/beacon-kit/beacon/payload-time"
-	"github.com/berachain/beacon-kit/config/spec"
 	engineprimitives "github.com/berachain/beacon-kit/engine-primitives/engine-primitives"
 )
 
-// sendPostBlockFCU sends a forkchoice update to the execution client.
+// sendPostBlockFCU sends a forkchoice update to the execution client after a
+// block is finalized.This function should only be used to notify
+// the EL client of the new head and should not request optimistic builds, as:
+// Optimistic clients already request builds in handleOptimisticPayloadBuild()
+// Non-optimistic clients should never request optimistic builds.
```

Comment:
q: this function is still called in its own goroutine. Is this necessary? Can we call it synchronously? @nidhi-singh02  @shotes